main.py

```diff
@@ -0,0 +1,520 @@
+                    self.changeMode(self.mode)
+                    self.linePoints.append((self.mx, self.my))
+
+                if self.mode == "spot":
+                    self.work("spot")
+
+        if event.type == pygame.MOUSEBUTTONUP:
+            if (self.mode == "scale" and self.areaMode == "rect") or self.mode == "area":
+                if len(self.linePoints) == 1:
+                    self.linePoints.append((self.cx, self.linePoints[0][1]))
+                    self.linePoints.append((self.cx, self.cy))
+                    self.linePoints.append((self.linePoints[0][0], self.cy))
+                    self.selectionComplete = True
+
+    def update(self, time_del):
+        """Update events."""
+        self.managers[self.mode].update(time_del)
+
+    def draw(self, surface):
+        """Draw contents on screen."""
+        surface.blit(self.background, (0, 0))
+        surface.blit(self.imsurf, (245, 15))
+        surface.blit(self.overlays, (245, 15))
+
+        pygame.draw.rect(surface, (255, 255, 255), (245, 540, 760, 35), 1)
+        self.managers[self.mode].draw_ui(surface)
+        surface.blit(
+            self.fonts[1].render(
+                f"x:{self.cx - 245:03}   y:{self.cy - 15:03}   temp:{self.mat_emm[self.cy - 15, self.cx - 245]:.4f}",
+                1,
+                (255, 255, 255),
+            ),
+            (253, 544),
+        )
+
+        if self.mode == "line":
+            if len(self.linePoints) == 1:
+                pygame.draw.line(surface, (255, 255, 255), self.linePoints[0], (self.cx, self.cy), 3)
+            if len(self.linePoints) == 2:
+                pygame.draw.line(surface, (255, 255, 255), self.linePoints[0], self.linePoints[1], 3)
+
+        if (self.mode == "scale" and self.areaMode == "poly") or self.mode == "emissivity":
+            if len(self.linePoints) > 0:
+                pygame.draw.lines(
+                    surface,
+                    (255, 255, 255),
+                    self.selectionComplete,
+                    self.linePoints + ([] if self.selectionComplete else [(self.cx, self.cy)]),
+                    3,
+                )
+        if (self.mode == "scale" and self.areaMode == "rect") or self.mode == "area":
+            if not self.selectionComplete:
+                if len(self.linePoints) > 0:
+                    pygame.draw.lines(
+                        surface,
+                        (255, 255, 255),
+                        True,
+                        self.linePoints
+                        + [(self.cx, self.linePoints[0][1]), (self.cx, self.cy), (self.linePoints[0][0], self.cy)],
+                        3,
+                    )
+            else:
+                pygame.draw.lines(surface, (255, 255, 255), True, self.linePoints, 3)
+
+        surface.blit(self.cursors[self.cursor_in], self.cursor_rect)
+
+
+if __name__ == "__main__":
+    pygame.mouse.set_visible(False)
+
+    pygame.display.set_caption("Detect Thermal Image Analysis")
```

Comment:
I know I recommended this earlier and sorry for the re-change request, but can you suffix a "Tool" at the end of the title?